`Acctest` - fix `mysql` RP test case failures

sinbai opened the pull request:
PR fixes the following test case failures in Teamcity:
1. `TestAccMySqlFlexibleServer_pitr` and `TestAccMySqlFlexibleServer_updateReplicationRole` fail with the following error, ignore_changes `source_server_id` because API does not return its value when `create_mode` is `PointInTimeRestore`/`Replica`.
![image](https://github.com/user-attachments/assets/94a10751-dfb7-4243-b1bc-ef74eaa73333)

> Refer to: PR #27295, PR #27442.

2. `TestAccDataSourceMySqlFlexibleServer_basic` fails with the following error, ignore_changes `zone` because API returns a value for it when it is not specified.
![image](https://github.com/user-attachments/assets/ee32b6e3-7f3c-45a9-b8a1-4c10a89881d5)


3. Some test cases failied with the following `ZoneNotAvailableForRegion` error, this is because the Available Zone of `eastus2` has changed from `1` and  `2` to `2`, `3`.
![image](https://github.com/user-attachments/assets/8983c3d4-35b0-460b-ba5e-5f3084d7c439)

4. Some test cases fail with the following  500 Internal Server Error, per [this ](https://learn.microsoft.com/zh-cn/azure/mysql/flexible-server/concepts-service-tiers-storage#service-tiers-size-and-server-types)document, the value ` B_Standard_B1s` of `sku_name` is currently not supported, change to `B_Standard_B1ms` instead of `B_Standard_B1s` to fix tests failure .
![image](https://github.com/user-attachments/assets/304c06d7-ca31-434f-853a-82746cf06a9c)

5. Fix test case `TestAccMySqlFlexibleServer_updateMaintenanceWindow` fails with the following issues.
![image](https://github.com/user-attachments/assets/0c2fab22-7d6e-4086-a99c-38a356cc0ba0)

![image](https://github.com/user-attachments/assets/a8112725-ad48-4120-9c8e-6a6511a96460)

> Note: Remove `zone` and ignore change `zone` since the `zone` was removed from func `basic` in [PR](https://github.com/hashicorp/terraform-provider-azurerm/pull/28799/files#diff-d24863cabe3a5a3e3e6acadc357ca64520901f1ffc25ce0bf48104065590a8b0L552), which caused error "zone cannot be changed independently". Given that removing `zone` from func `basic` is correct, and Maintenance Window testing does not require `zone` to be specified, so remove `zone` and ignore change `zone` to fix the failure. 

Test Results:
https://hashicorp.teamcity.com/buildConfiguration/TF_AzureRM_AZURERM_SERVICE_PUBLIC_MYSQL?branch=refs%2Fpull%2F28882%2Fmerge&buildTypeTab=overview

stephybun reviewed (changes requested):
@sinbai could you resolve the merge conflict and also remove `zone` from `ignore_changes` in the tests, we introduced a Computed schema for zone which should fix those tests.

sinbai commented:
> @sinbai could you resolve the merge conflict and also remove `zone` from `ignore_changes` in the tests, we introduced a Computed schema for zone which should fix those tests.

Hi @stephybun thanks for your feedback. I have updated the code and all the tests have passed in Teamcity: https://hashicorp.teamcity.com/buildConfiguration/TF_AzureRM_AZURERM_SERVICE_PUBLIC_MYSQL?branch=refs%2Fpull%2F28882%2Fmerge&buildTypeTab=overview, could you please take another look?

stephybun reviewed (approved):
Thanks @sinbai LGTM 👍
